query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/simple/find_unique.rs

```diff
@@ -89,4 +89,141 @@ mod find_unique {
 
         Ok(())
     }
+
+    fn simple_uniq_idx_with_embedded() -> String {
+        indoc! {r#"
+        type Location {
+            address String
+        }
+
+        model A {
+            #id(id, Int, @id)
+            name String
+            location Location
+
+            @@unique([location.address])
+        }
+        "#}
+        .to_string()
+    }
+
+    #[connector_test(schema(simple_uniq_idx_with_embedded), only(MongoDb))]
+    async fn simple_embedded_type(runner: Runner) -> TestResult<()> {
+        run_query!(
+            runner,
+            indoc! {r#"mutation {
+            createManyA(data: [
+                {id: 1 name: "foo" location: {set: {address: "a"}}},
+                {id: 2 name: "foo" location: {set: {address: "b"}}},
+                {id: 3 name: "foo" location: {set: {address: "c"}}},
+            ]) { count }
+        }"#}
+        );
+
+        assert_query!(
+            runner,
+            r#"query { findUniqueA(where: { 
+                location_address: {
+                    location: {
```

Comment:
Can you add a test for using `name` with `@@unique`:

```rust
fn composite_uniq_idx_with_embedded() -> String {
        indoc! {r#"
        type Location {
            address Int
        }

        model A {
            #id(id, Int, @id)
            name String
            location Location

            @@unique([name, location.address], name: "name_address")
        }
        "#}
        .to_string()
    }
 ```

And run a query like so:

```rust
assert_query!(
            runner,
            r#"query { findUniqueA(where: { 
                name_address: {
                    name: "foo"
                    location: {
                        address: 1
                    }
                } 
            }) { id }}"#,
            r#"{"data":{"findUniqueA":{"id":1}}}"#
        );
```
   
   
   Sadly we are also missing a test for non-composites with the `name = ...` check. Could you add that in as well.

Reply:
I actually agree but **not** in this test case. This should be a test around the general usage of `find unique` not adding composites to the equation. I checked and there are no tests using named indices so I think that should be added in that file (assuming I will split this in a separate file). Once added there it will be enough. Remember, we are testing the case of covering embedded documents in a unique index, not the workings of find unique.

Reply:
I will add that as an additional test to find_unique after this is merged (is not part of what needs to be cover in this PR)